### PR TITLE
chore: update the readme

### PR DESCRIPTION
**Pull request by irbull**
Update the README.md file to reflect the current state of the project. This includes adding a section on how to build and contribute to the project and updates the note about who is currently maintaining the project.

This change-set also removes the CHANGELOG.md file as most Deno projects don't include this and it was mostly a change log of the Astro project.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -190,22 +191,28 @@ The [Astro Deno](https://github.com/withastro/astro/tree/main/examples/deno)
...
-
-For help, check out the `#support` channel on
-[Discord](https://astro.build/chat). Our friendly Support Squad members are here
-to help!
+## Contributing
 
-You can also check our [Astro Integration Documentation][astro-integration] for
-more on integrations.
+To configure your development environment, clone the repository and install
+[`pnpm`](https://pnpm.io/). `pnpm` is a package manager that emphasizes disk
```
Comment (kt3k) [inline README.md:197]: Do you prefer to use pnpm? I'm personally in favor of getting rid of pnpm settings (Currently the repo has both `pnpm-lock.yaml` and `package-lock.json` and that feels confusing to me)

Comment (irbull) [reply README.md:197]: There is also a `deno.lock` file ;). I guess we don't need the `package-lock.json`? Should we remove that one, assuming that we really do need `pnpm` to build this project right now?

Comment (kt3k) [reply README.md:197]: Ok. Let's revisit that in later PRs

**Review by kt3k — APPROVED** [review]
LGTM

**Comment by und-miller** [comment]
NPM link to the changelog should be removed: https://www.npmjs.com/package/@deno/astro-adapter#changelog

**Comment by irbull** [comment]
> NPM link to the changelog should be removed: https://www.npmjs.com/package/@deno/astro-adapter#changelog

Good catch. Actually that entire page is out-of-date. I've updated the README, but it's not reflected there. I wonder who has access to update the NPM description.

**Pull request merged**